### PR TITLE
navigator: Don't clean already existing snapshots when running navigate_and_compare()

### PR DESCRIPTION
**Pull request by xchapron-ledger**

### CODE REVIEW

File: src/ragger/navigator/navigator.py [hunk 1]
```diff
@@ -131,11 +131,18 @@ def _check_snaps_dir_path(self, path: Path, test_case_name: Path, is_golden: boo
                 raise ValueError(f"Golden snapshots directory ({dir_path}) does not exist.")
         return dir_path
 
-    def _init_snaps_temp_dir(self, path: Path, test_case_name: Path) -> Path:
+    def _init_snaps_temp_dir(self, path: Path, test_case_name: Path, star_idx: int = 0) -> Path:
```
Comment (fbeutin-ledger) [inline src/ragger/navigator/navigator.py:134]: start*

File: src/ragger/navigator/navigator.py [hunk 2]
```diff
@@ -131,11 +131,18 @@ def _check_snaps_dir_path(self, path: Path, test_case_name: Path, is_golden: boo
...
             for file in snaps_tmp_path.iterdir():
-                file.unlink()
+                if not file.name.endswith(".png"):
+                    continue
+                file_idx_str = file.name.replace(".png", "")
+                if not file_idx_str.isnumeric():
+                    continue
+                file_idx = int(file_idx_str)
+                if file_idx >= star_idx:
+                    file.unlink()
```
Comment (fbeutin-ledger) [inline src/ragger/navigator/navigator.py:145]: Add comments to explain what this does and what the parameter `start_idx` means

**Review by fbeutin-ledger — APPROVED** [review]

File: src/ragger/navigator/navigator.py (hunk 2 above)
Comment (agrojean-ledger) [reply src/ragger/navigator/navigator.py:145]: My understanding is that the function removes snap files with an index > start index only

Comment (xchapron-ledger) [reply src/ragger/navigator/navigator.py]: Comment added :)

**Review by agrojean-ledger — APPROVED** [review]

**Pull request merged**